File: flut/.flutter/lib/flutter/widgets/icon_theme_data.dart
import 'package:flutter/material.dart';
import 'package:flut/flut/runtime.dart';
import 'package:flut/flut/object.dart';
import 'package:flut/dart/ui.dart';

class FlutIconThemeData extends FlutValueObject {
  final IconThemeData value;
  const FlutIconThemeData(this.value) : super('IconThemeData');

  @override
  Map<String, dynamic> flutEncode() {
    final result = flutBaseProps();
    if (value.size != null) result['size'] = value.size;
    if (value.fill != null) result['fill'] = value.fill;
    if (value.weight != null) result['weight'] = value.weight;
    if (value.grade != null) result['grade'] = value.grade;
    if (value.opticalSize != null) result['opticalSize'] = value.opticalSize;
    if (value.color != null) {
      result['color'] = FlutColor(value.color!).flutEncode();
    }
    if (value.opacity != null) result['opacity'] = value.opacity;
    if (value.shadows != null) {
      result['shadows'] = value.shadows!
          .map((s) => FlutShadow(s).flutEncode())
          .toList();
    }
    if (value.applyTextScaling != null) {
      result['applyTextScaling'] = value.applyTextScaling;
    }
    return result;
  }

  static IconThemeData? flutDecode(
    FlutRuntime runtime,
    Map<String, dynamic> data,
  ) {
    return IconThemeData(
      size: runtime.unpackOptionalField<double>(data, 'size'),
      fill: runtime.unpackOptionalField<double>(data, 'fill'),
      weight: runtime.unpackOptionalField<double>(data, 'weight'),
      grade: runtime.unpackOptionalField<double>(data, 'grade'),
      opticalSize: runtime.unpackOptionalField<double>(data, 'opticalSize'),
      color: runtime.unpackOptionalField<Color>(data, 'color'),
      opacity: runtime.unpackOptionalField<double>(data, 'opacity'),
      shadows: runtime.unpackOptionalField<List<Shadow>>(data, 'shadows'),
      applyTextScaling: runtime.unpackOptionalField<bool>(
        data,
        'applyTextScaling',
      ),
    );
  }
}

File: flut/flutter/widgets/icon_theme_data.py
from typing import List, Optional, override

from flut._flut.engine import FlutValueObject, _flut_pack_value
from flut._flut.runtime import _flut_unpack_optional_field
from flut.dart.ui import Color


class IconThemeData(FlutValueObject):
    _flut_type = "IconThemeData"

    def __init__(
        self,
        *,
        size: Optional[float] = None,
        fill: Optional[float] = None,
        weight: Optional[float] = None,
        grade: Optional[float] = None,
        opticalSize: Optional[float] = None,
        color: Optional[Color] = None,
        opacity: Optional[float] = None,
        shadows: Optional[List] = None,
        applyTextScaling: Optional[bool] = None,
    ):
        super().__init__()
        self.size = size
        self.fill = fill
        self.weight = weight
        self.grade = grade
        self.opticalSize = opticalSize
        self.color = color
        self.opacity = opacity
        self.shadows = shadows
        self.applyTextScaling = applyTextScaling

    @override
    def _flut_pack(self) -> dict:
        result = self._flut_base_props()
        if self.size is not None:
            result["size"] = _flut_pack_value(self.size)
        if self.fill is not None:
            result["fill"] = _flut_pack_value(self.fill)
        if self.weight is not None:
            result["weight"] = _flut_pack_value(self.weight)
        if self.grade is not None:
            result["grade"] = _flut_pack_value(self.grade)
        if self.opticalSize is not None:
            result["opticalSize"] = _flut_pack_value(self.opticalSize)
        if self.color is not None:
            result["color"] = _flut_pack_value(self.color)
        if self.opacity is not None:
            result["opacity"] = _flut_pack_value(self.opacity)
        if self.shadows is not None:
            result["shadows"] = _flut_pack_value(self.shadows)
        if self.applyTextScaling is not None:
            result["applyTextScaling"] = _flut_pack_value(self.applyTextScaling)
        return result

    @staticmethod
    def _flut_unpack(data: dict) -> "IconThemeData":
        return IconThemeData(
            size=_flut_unpack_optional_field(data, "size"),
            fill=_flut_unpack_optional_field(data, "fill"),
            weight=_flut_unpack_optional_field(data, "weight"),
            grade=_flut_unpack_optional_field(data, "grade"),
            opticalSize=_flut_unpack_optional_field(data, "opticalSize"),
            color=_flut_unpack_optional_field(data, "color"),
            opacity=_flut_unpack_optional_field(data, "opacity"),
            shadows=_flut_unpack_optional_field(data, "shadows"),
            applyTextScaling=_flut_unpack_optional_field(data, "applyTextScaling"),
        )
